components/test_pd/src/server.rs

```diff
@@ -92,7 +92,10 @@ impl<C: PdMocker + Send + Sync + 'static> Server<C> {
     }
 
     pub fn stop(&mut self) {
-        self.server.take();
+        self.server
+            .take()
+            .expect("Server is not started")
+            .shutdown();
```

Comment:
if we just drop server instead of shutdown, there is 50% chance that the server after startup will make subsequent grpc call cancelled. I don’t know the specific reason, but I think it is reasonable to use shutdown when stopping the server. @BusyJay